components/net/connector.rs

```diff
@@ -28,33 +29,18 @@ const DEFAULT_CIPHERS: &'static str = concat!(
 );
 
 pub fn create_http_connector() -> Arc<Pool<Connector>> {
-    let mut context = SslContext::new(SslMethod::Sslv23).unwrap();
-    context.set_CA_file(&resources_dir_path()
-                        .expect("Need certificate file to make network requests")
-                        .join("certs")).unwrap();
-    context.set_cipher_list(DEFAULT_CIPHERS).unwrap();
-    context.set_options(SSL_OP_NO_SSLV2 | SSL_OP_NO_SSLV3 | SSL_OP_NO_COMPRESSION);
-    let connector = HttpsConnector::new(ServoSslClient {
-        context: Arc::new(context)
-    });
-
-    Arc::new(Pool::with_connector(Default::default(), connector))
-}
-
-pub struct ServoSslClient {
-    context: Arc<SslContext>,
-}
-
-impl SslClient for ServoSslClient {
-    type Stream = SslStream<HttpStream>;
-
-    fn wrap_client(&self, stream: HttpStream, host: &str) -> Result<Self::Stream, ::hyper::Error> {
-        let mut ssl = try!(Ssl::new(&self.context));
-        try!(ssl.set_hostname(host));
-        let host = host.to_owned();
-        ssl.set_verify_callback(SSL_VERIFY_PEER, move |p, x| {
-            ::openssl_verify::verify_callback(&host, p, x)
-        });
-        SslStream::connect(ssl, stream).map_err(From::from)
+    let ca_file = &resources_dir_path()
+        .expect("Need certificate file to make network requests")
+        .join("certs");
+    let mut ssl_connector_builder = SslConnectorBuilder::new(SslMethod::tls()).unwrap();
+    {
+        let ssl_context_builder = ssl_connector_builder.builder_mut();
+        ssl_context_builder.set_ca_file(ca_file).expect("could not set CA file");
+        ssl_context_builder.set_cipher_list(DEFAULT_CIPHERS).expect("could not set ciphers");
+        ssl_context_builder.set_options(SSL_OP_NO_SSLV2 | SSL_OP_NO_SSLV3 | SSL_OP_NO_COMPRESSION);
```

Comment:
These are all set already by the connector fyi: https://github.com/sfackler/rust-openssl/blob/master/openssl/src/ssl/connector.rs#L29